### ISSUE TITLE
Payment page broken on cbs.com

### ISSUE THREAD

**Issue by remusao** [uBlockOrigin/uAssets#4908]
Hi all,

### URL(s) where the issue occurs

Credentials are needed to access the payment page but the URLs is: `https://www.cbs.com/all-access/subscription/payment/`

I put few screenshots at the end to show the issue.

### Describe the issue

The commit https://github.com/uBlockOrigin/uAssets/commit/4073ed711890baf7a03ef571988397bb4b84c6f4 introduced the following filter for domain cbs.com: `*/*$script,redirect=noopjs,domain=cbs.com,third-party`. It is breaking payment since it blocks any third-party script, which seems a bit aggressive (including: `https://js.recurly.com/v4/recurly.js` needed for payment).

### Screenshot(s)

Here is the payment page where the missing credit cart information fields are missing (highlighted in red):

![cbs_payment_missing](https://user-images.githubusercontent.com/1299873/52778848-31555d00-3047-11e9-95c9-8bfaed415e8e.png)

From the logger we see that `https://js.recurly.com/v4/recurly.js` is blocked (actually it is redirected to `noopjs`):

![cbs_request_blocked1](https://user-images.githubusercontent.com/1299873/52778860-374b3e00-3047-11e9-864a-8f7d6052c14e.png)

After applying the following custom rule: `@@https://js.recurly.com/v4/recurly.js$script,domain=cbs.com,third-party` it works.

![cbs_payment_unbroken](https://user-images.githubusercontent.com/1299873/52778898-4f22c200-3047-11e9-995c-5df0194f44fc.png)

The issue was also resolved if I use the following custom rule: `*/*$third-party,script,domain=cbs.com,badfilter`

Maybe this filter could be made more specific to avoid blocking all third-party scripts?

### Versions

- Browser/version: seems to happen in any browser (issue due to filter)
- uBlock Origin version: v1.18.4

### Settings

- Using the list `ublock-filters` is enough to reproduce this issue

### Notes

Culprit filter: `*/*$script,redirect=noopjs,domain=cbs.com,third-party`
Fix involves either whitelisting the specific third-party script needed for payment with `@@https://js.recurly.com/v4/recurly.js$script,domain=cbs.com,third-party` or remove the filter completely with `*/*$third-party,script,domain=cbs.com,badfilter`.

Thank you very much!

**Comment by okiehsch** [comment]
I don't need `*/*$third-party,script,domain=cbs.com` to watch https://github.com/uBlockOrigin/uAssets/issues/4651
````
https://www.cbs.com/shows/ncis/video/bDxO0_npmKpYWsgBKszGAFMLIJldDFOI/ncis-the-last-link
````
so the filter can be removed as far as I am concerned.
Seems a bit to aggressive for such a site anyway.

@mapx- do you need the filter?

**Comment by mapx-** [comment]
it was for anti adb stuff. I'd add that specific whitelisting filter.

**Comment by remusao** [comment]
Thanks a lot for the quick response time. I was a bit confused by this particular filter, if the intent was to block all third-party scripts, why not having this filter instead: `*$script,third-party,redirect=noopjs,domain=cbs.com`?

**Comment by mapx-** [comment]
You mean why 
`*/*$script,redirect=noopjs,domain=cbs.com,third-party`
and not 
`*$script,third-party,redirect=noopjs,domain=cbs.com`
?

The second filter (see in your logger) ignore the redirection

**Comment by remusao** [comment]
Yes that's what I meant, and in latest uBlock Origin at least they seem to be equivalent. I was able to disable: `*/*$script,redirect=noopjs,domain=cbs.com,third-party` and replace it with `*$script,third-party,redirect=noopjs,domain=cbs.com` with the same requests being redirected.

**Comment by mapx-** [comment]
Weird, I tested right now in FF 65.0.1 and I see the same behaviour like before.

**Comment by remusao** [comment]
Mmh Ok now I am confused and not sure if we are talking about the same thing. My point was that `*/*$script,redirect=noopjs,domain=cbs.com,third-party` and `*$script,third-party,redirect=noopjs,domain=cbs.com` are equivalent. But I found the second one to express more clearly the intent of "block all third-party scripts on cbs.com". In short, I did not understand why using `*/*` in the pattern since it will match any URL, just like `*` would.

In any case, the current fix worked for resolving the payment issue. Thanks a lot for that!
